### ISSUE TITLE
Unnecessary `=""` added

### ISSUE THREAD

**Issue by Turbo87** [ember-template-lint/ember-template-recast#180]
This is related to https://github.com/ember-codemods/ember-no-implicit-this-codemod/issues/53

In some situation the printer is adding `=""` at the end of a named argument for some reason.

Example:
```hbs
<SomeComponent
  class=""
  @foo=42
  @bar={{action this.baz}}
/>
```

with

```js
MustacheStatement(node) {
  node.params = node.params.slice();
},
```

results in: 
```hbs
<SomeComponent
  class=""
  @foo=42=""
  @bar={{action this.baz}}
/>
```

/cc @kevinansfield

**Comment by rwjblue** [comment]
WAT

**Comment by Turbo87** [comment]
🤷‍♂ 

left you a reprodution in #181 😉

**Comment by rwjblue** [comment]
A few things to note here:

1. Using `@foo=42` will actually result in passing `"42"` (because all attributes / arguments that are passed without curlies are strings)
2. We are incorrectly detecting that the "join open parts with" (basically the whitespace + indentation that we should use to add any additional params with) is working improperly with the AttrNode + TextNode with empty chars fix that we have.
3. I pushed commits to fix the issue over in #181, will land and release once CI is happy.

**Comment by rwjblue** [comment]
The reason it presented as it did (`@foo=42=""`), was that we use the distance between the end of the first attribute and the start of the second one to decide how much whitespace to add automatically. This ensures that if you add an additional `AttrNode` or `ElementModifierStatement` it will follow the formatting that you previously had. In this case, the `AttrNode` was incorrectly assumed to be "valueless" and therefore the end location was "corrected" to an invalid value, this resulted in the whitespace auto-detection thinking that the value to join with was `=""\n  ` 😩.

**Comment by rwjblue** [comment]
Closed by #181
